tests/cupyx_tests/test_optimize.py

```diff
@@ -120,3 +121,60 @@ def test_optimize_pickle(self):
             with cupyx.optimizing.optimize(key='other_key') as context:
                 with pytest.raises(ValueError):
                     context.load(filepath)
+
+
+# TODO(leofang): check the optimizer is not applicable to the cutensor backend?
+@testing.parameterize(*testing.product({
+    'backend': ([], ['cub'])
+}))
+@testing.gpu
+@testing.with_requires('optuna')
+class TestOptimizeBackends(unittest.TestCase):
+    """This class tests if optuna is in effect for create_reduction_func()"""
+
+    def setUp(self):
+        cupy.core._optimize_config._clear_all_contexts_cache()
+        self.old_reductions = _accelerator.get_reduction_accelerators()
+        _accelerator.set_reduction_accelerators(self.backend)
+
+        # avoid shadowed by the cub module
+        self.old_routines = _accelerator.get_routine_accelerators()
+        _accelerator.set_routine_accelerators([])
+
+        self.x = testing.shaped_arange((3, 4), cupy, dtype=cupy.float32)
+
+    def tearDown(self):
+        _accelerator.set_routine_accelerators(self.old_routines)
+        _accelerator.set_reduction_accelerators(self.old_reductions)
+
+    def test_optimize1(self):
+        # Ensure the optimizer is run 3 times for all backends.
+        func = 'cupyx.optimizing._optimize._optimize'
+        is_called = 3
+
+        # Setting "wraps" is necessary to avoid compilation errors.
+        with testing.AssertFunctionIsCalled(
```

Comment:
why not directly use `mock.patch` like the examples above?

Reply:
Because I like my `AssertFunctionIsCalled` class 😁 Another reason is that I tested the aggregated counts to avoid calling `assert` multiple times. Let me know your preference.

Reply:
I think this is fine, I was just curious 😁